client/state/jetpack-connect/reducer.js

```diff
@@ -126,6 +127,18 @@ export function jetpackConnectSite( state = {}, action ) {
 	return state;
 }
 
+export function jetpackConnectSitesList( state = {}, action ) {
```

Comment:
Yes. I'd like to keep it in a new reducer tree. When we we dispatch JETPACK_CONNECT_COMPLETE_FLOW, i'd like jetpackConnectAuthorize to be an empty state. However, because of the requirements described above, in some cases we'll need to know when a new site is connected. That's why I added this to a separate branch.